### PR TITLE
adds open issues section

### PR DESCRIPTION
**Pull request by cc10512**
And cleans up the formatting so that we minimize the number of
warnings and errors.

### CODE REVIEW

File: p4-16/psa/PSA.mdk [hunk 1]
```diff
@@ -2158,7 +2158,50 @@ The control plane should provide a
...
+may happen.
+
+In the current specification, each extern has its own resources and they are constrained to a
+single pipeline, i.e., not shared across pipelines and not maintained coherent with any other
+externs even if attached to the same logical table. It is the responsability of the programmer to
+maintain state across pipelines using control-plane operations, if so desired.
+
+Likewise, for programming multiple pipelines, it is the responsability of the vendor and of target
+dependent tools to specify how PSA programs are mapped to multiple pipelines. A simple
+implementation may use a copy of the PSA program on each pipeline, and thus keeping pipelines full
```
Comment (jafingerhut) [inline p4-16/psa/PSA.mdk]: full -> fully

File: p4-16/psa/psa.p4 [hunk 2]
```diff
@@ -638,7 +640,11 @@ package PSA_Switch<IH, IM, EH, EM> (IngressPipeline<IH, IM> ingress,
 // constructor calls for PacketReplicationEngine and
 // BufferingQueueingEngine.
 
-#define PSA_SWITCH(ip, ep) PSA_Switch((ip), PacketReplicationEngine(), (ep), BufferingQueueingEngine())
+#define PSA_SWITCH(ip, ep) PSA_Switch(                           \
+                                      (ip),                      \
+                                      PacketReplicationEngine(), \
+                                      (ep),                      \
+                                      BufferingQueueingEngine())
 // END:Programmable_blocks
```
Comment (jafingerhut) [inline p4-16/psa/psa.p4]: We could consider moving this line '// END:Programmable_blocks' up to just before the macro is defined, if we are not sure we want to include it in the PSA spec itself, e.g. if the goal is to remove it.

File: p4-16/psa/PSA.mdk [hunk 3]
```diff
@@ -2158,7 +2158,50 @@ The control plane should provide a
...
+
+## Action Selectors
+
+The size parameter in the action_selector instance that defines the maximum number of members in a
+selector. In some cases it might be useful to allow the controller to dynamically provision
+resources on the selector or to utilize different selector sizes on different targets, while using
+a common P4 program.
+
+We also need to formalize the interaction of action profiles and action selectors with counters and
+meters.
```
Comment (jafingerhut) [inline p4-16/psa/PSA.mdk]: might want to qualify by saying 'action selectors with direct counters and meters'.  I don't think there are any open questions about interactions between action selectors and indexed counters and meters.

File: p4-16/psa/PSA.mdk [hunk 4]
```diff
@@ -2135,7 +2135,7 @@ digest message can contain any from the dataplane.
 
 In PSA, digest is performed at the end of the ingress pipeline in the
 deparser. The digest extern provides a `pack` method to specify the
-content in the 
+content in the
```
Comment (jafingerhut) [inline p4-16/psa/PSA.mdk]: Might be good to add an explicit TBD here to remind ourselves of this incomplete sentence.  Better of course to actually complete it.

**Comment by cc10512** [comment]
I fixed the issues you pointed out.

**Pull request merged**